Bundler/Viewer/Unity/Assets/Scripts/Network/Scripts/NetworkAnchorManager.cs

```diff
@@ -0,0 +1,487 @@
+        {
+            Debug.LogErrorFormat("[NetworkAnchorManager] Failed to receive anchor data. {0}", DebugInfo());
+            ImportAnchorDataCompleted(null);
+            return;
+        }
+
+        if (args.Data == null || args.Data.Length == 0)
+        {
+            Debug.LogErrorFormat("[NetworkAnchorManager] Binary anchor data is null or empty, ignoring request to import anchor data. {0}", DebugInfo());
+            ImportAnchorDataCompleted(null);
+            return;
+        }
+
+        Debug.LogFormat("[NetworkAnchorManager] Starting import of binary anchor data. (bytes: {0}) {1}", args.Data.Length, DebugInfo());
+        WorldAnchorTransferBatch.ImportAsync(args.Data, BatchImportAsyncCompleted);
+    }
+
+    /// <summary>
+    /// Called when a remote anchor has been de-serialized
+    /// </summary>
+    /// <param name="status">Tracks if the import worked</param>
+    /// <param name="batch">The WorldAnchorTransferBatch that has the anchor information.</param>
+    private void BatchImportAsyncCompleted(
+        SerializationCompletionReason status,
+        WorldAnchorTransferBatch batch)
+    {
+        if (status != SerializationCompletionReason.Succeeded)
+        {
+            Debug.LogErrorFormat("[NetworkAnchorManager] Anchor import has failed. (status: {0})", status);
+            ImportAnchorDataCompleted(null);
+        }
+        else
+        {
+            Debug.Log("[NetworkAnchorManager] Anchor import was successful.");
+            ImportAnchorDataCompleted(batch);
+        }
+    }
+
+    /// <summary>
+    /// The final function called once a network anchor has been imported.
+    /// </summary>
+    private void ImportAnchorDataCompleted(WorldAnchorTransferBatch batch)
+    {
+        if (batch == null)
+        {
+            LastReceivedAnchor = null;
+        }
+        else
+        {
+            LastReceivedAnchor = new LastReceivedAnchorArgs(batch);
+        }
+
+        LoadingAnchor = false;
+
+        if (LastReceivedAnchorChanged != null)
+        {
+            LastReceivedAnchorChanged(this, LastReceivedAnchor);
+        }
+    }
+#endregion Importing Anchor Methods
+
+#region Initialization Methods
+    /// <summary>
+    /// On creation save this as the static instance. There should be only one manager.
+    /// </summary>
+    private void Awake()
+    {
+        DontDestroyOnLoad(gameObject);
+        instance = this;
+
+        this.InitializeLocalAddress();
+        this.InitializeAnchorTransmitterOnce();
+    }
+
+    /// <summary>
+    /// Initialize the local IP address. This should only be called with authority.
+    /// </summary>
+    private void InitializeLocalAddress()
+    {
+#if !UNITY_EDITOR
+        foreach (Windows.Networking.HostName hostName in Windows.Networking.Connectivity.NetworkInformation.GetHostNames())
+        {
+            if (hostName.DisplayName.Split(".".ToCharArray()).Length == 4)
+            {
+                LocalAddress = hostName.DisplayName;
```

Comment:
This feels odd... to me, an address is an ipv4 or ipv6 address. what happens if a host doesn't support network resolvable names? do you get a local name or the ip address?

Reply:
"pending" good questions.  This code was copied from Holo Academy.  Is there are better way to obtain the client's IP address?

Reply:
I just noticed the assumption that hostName.DisplayName is a string in IPv4 format. i'd have to dig back into my brain for my old .net tcp/sockets knowledge... for now, this is probably ok, i'd just add a comment pointing out the assumption, just in case someone trips on it.

Reply:
Ok...leaving as is for now.  Added an LogError on a failure